src/tools/chip-cert/CertUtils.cpp

```diff
@@ -310,6 +330,108 @@ bool AddExtension(X509 * cert, int extNID, const char * extStr)
     return res;
 }
 
+bool SetBasicConstraintsExtension(X509 * cert, bool isCA, int pathLen, CertStructConfig & certConfig)
+{
+    if (certConfig.IsExtensionBasicPresent())
```

Comment:
nit: some indentation would decrease (and readability increase) if you invert conditions:

```cpp
if (!certConfig.IsExtensionBasicPresent()) {
  return true;
}

// ....

return AddExtension(cert, NID_basic_constraints, basicConstraintsExt.c_str());
```

Reply:
done